tests/performance/density-kwok.go

```diff
@@ -0,0 +1,322 @@
+/*
+Copyright 2024 The KubeVirt Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package performance
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"time"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	k8sv1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	v1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/kubecli"
+
+	"kubevirt.io/kubevirt/pkg/libvmi"
+	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+	"kubevirt.io/kubevirt/tests/libvmifact"
+	"kubevirt.io/kubevirt/tests/testsuite"
+)
+
+const (
+	vmBatchStartupLimit = 5 * time.Minute
+	defaultNodeCount    = 100
+	defaultVMCount      = 1000
+)
+
+var (
+	nodeCount = getNodeCount()
+	vmCount   = getVMCount()
+)
+
+var _ = Describe("Control Plane Performance Density Testing using kwok", Label("KWOK", "sig-performance", "Serial"), Serial, Ordered, func() {
+	var (
+		kubevirtClient kubecli.KubevirtClient
+		startTime      time.Time
+	)
+
+	artifactsDir, _ := os.LookupEnv("ARTIFACTS")
+
+	BeforeEach(func() {
+		// Check if required labels are present in the label filter
+		labelFilter := os.Getenv("KUBEVIRT_E2E_LABEL_FILTER")
+		requiredLabels := []string{"KWOK", "sig-performance"}
+
+		if !labelsMatchFilter(labelFilter, requiredLabels) {
+			Skip("Required labels not present in the label filter")
+		}
```

Comment:
Shouldn't this be controlled by Ginkgo, without custom code?

Reply:
Do you mean via skip and focus? since other performance tests are using env variables and not using skip and focus to take a decision and run tests, this tests would also run and panic.